Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -993,18 +996,66 @@ private PointerData GetFirstPointer()
 
         private bool TryGetGripRotation(IMixedRealityPointer pointer, out Quaternion rotation)
         {
-            for (int i = 0; i < (pointer.Controller?.Interactions?.Length ?? 0); i++)
+            rotation = Quaternion.identity;
+            switch (pointer.Controller.ControllerHandedness)
             {
-                if (pointer.Controller.Interactions[i].InputType == DeviceInputType.SpatialGrip)
-                {
-                    rotation = pointer.Controller.Interactions[i].RotationData;
-                    return true;
-                }
+                case Handedness.Left:
+                    rotation = LeftHandRotation;
+                    break;
+                case Handedness.Right:
+                    rotation = RightHandRotation;
+                    break;
+                default:
+                    return false;
+            }
+            return true;
+        }
+
+        #endregion
+
+        #region Source Pose Handler Implementation
+        /// <summary>
+        /// Raised when the source pose tracking state is changed.
+        /// </summary>
+        public void OnSourcePoseChanged(SourcePoseEventData<TrackingState> eventData) { }
+
+        /// <summary>
+        /// Raised when the source position is changed.
+        /// </summary>
+        public void OnSourcePoseChanged(SourcePoseEventData<Vector2> eventData) { }
+
+        /// <summary>
+        /// Raised when the source position is changed.
+        /// </summary>
+        public void OnSourcePoseChanged(SourcePoseEventData<Vector3> eventData) { }
+
+        /// <summary>
+        /// Raised when the source rotation is changed.
+        /// </summary>
+        public void OnSourcePoseChanged(SourcePoseEventData<Quaternion> eventData) { }
+
```

Comment:
Completely unrelated and no change needed here, but strong dislike on how many different methods you had to implement just to get `IMixedRealitySourcePoseHandler` working! We should really consider using generic typing like we do for https://github.com/microsoft/MixedRealityToolkit-Unity/blob/0c454151a7e285382b3485d19052c86a0f5198b0/Assets/MRTK/Core/Interfaces/InputSystem/Handlers/IMixedRealityInputHandler.cs#L30

Reply:
Issue to track this code improvement here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/9256 

Probably going to avoid rocking that boat on this PR

Reply:
>Probably going to avoid rocking that boat on this PR

100%!